cmd/kubeadm/app/apis/kubeadm/v1beta2/doc.go

```diff
@@ -23,252 +23,271 @@ limitations under the License.
+// bootstrapTokens:
+//   # ...
+// nodeRegistration:
+//   # ...
+// ```
+//
+// The `InitConfiguration` type is used to configure runtime settings. In the case of `kubeadm init`,
+// it includes the configuration of the bootstrap token and all the setting specific to the node
+// where kubeadm is executed, including:
+//
+// - `nodeRegistration`: fields that relate to registering the new node to the cluster.
+//   You can use it to customize the node name, the CRI socket to use or any other
+//   settings that should apply to this node only (for example. the node IP).
+//
+// - `localAPIEndpoint`: the endpoint of the API server instance to be deployed on this node.
+//   For example, you can use it to customize the API server advertise address.
+//
+//   ```yaml
+//   apiVersion: kubeadm.k8s.io/v1beta2
+//   kind: ClusterConfiguration
+//   networking:
+//     # ...
+//   etcd:
+//     # ...
+//   apiServer:
+//     extraArgs:
+//       # ...
+//     extraVolumes:
+//       # ...
+//   # ...
+//   ```
+//
+// The `ClusterConfiguration` type can be used to configure cluster-wide settings, including:
+//
+// - Networking, that holds configuration for the networking topology of the cluster;
+//   For example, uou can use it to customize node subnet or services subnet.
+//
+// - Etcd configurations that can be used for customizing the local etcd or to configure
+//   the API server for using an external etcd cluster.
+//
+// - kube-apiserver, kube-scheduler, kube-controller-manager configurations.
+//   You can use it to customize control-plane components by adding customized setting
+//   or overriding kubeadm default settings.
+//
+// ```yaml
+// apiVersion: kubeproxy.config.k8s.io/v1alpha1
+// kind: KubeProxyConfiguration
+// # ...
+// ```
+//
+// The `KubeProxyConfiguration` type is used to change the configurations passed to the kube-proxy
+// instances deployed in the cluster. If this object is not provided or provided only partially,
+// kubeadm applies defaults.
+// See [kube-proxy reference](https://kubernetes.io/docs/reference/command-line-tools-reference/kube-proxy/)
+// or [kube-proxy source code](https://godoc.org/k8s.io/kube-proxy/config/v1alpha1#KubeProxyConfiguration)
+// for the official documentation.
+//
+// ```yaml
+// apiVersion: kubelet.config.k8s.io/v1beta1
+// kind: KubeletConfiguration
+// # ...
+// ```
+//
+// The `KubeletConfiguration` type is used to change the configurations passed to all kubelet instances
 // deployed in the cluster. If this object is not provided or provided only partially, kubeadm applies defaults.
-//
-// See https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/ or https://godoc.org/k8s.io/kubelet/config/v1beta1#KubeletConfiguration
-// for kubelet official documentation.
+// See [kubelet reference](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/) or
+// [kubelet source code](https://godoc.org/k8s.io/kubelet/config/v1beta1#KubeletConfiguration)
```

Comment:
`[kubelet source code]` it's technically not the kubelet source code.
we could call it `[kubelet API documentation]`.

Reply:
Right. Those are godocs, not Go source. How about just call it "GoDoc"?
Calling it "kubelet API reference" isn't accurate either.

Reply:
GoDoc SGTM